refresh-data.sh

```diff
@@ -10,22 +10,19 @@
 
 BRANCH=$1
 
-if [ "$BRANCH" == "" ]; then
-  echo "Specify a branch name as an argument"
-  exit 1
-fi
-
 if [ "$BRANCH" == "master" ]; then
   echo "Doing this to master would be a terrible idea"
   exit 1
 fi
 
-git fetch origin
-git reset --hard
-git checkout origin/master
-git reset --hard
-if git rev-parse --verify $BRANCH ; then
-    git branch -D $BRANCH
+if [ "$(git status --porcelain)" ]; then
+   echo "git status must be clean in $(pwd)"
+   exit 1
+fi
+
+if [ "$BRANCH" ]; then
+  git fetch origin
+  git checkout -B "$BRANCH" origin/master
```

Comment:
There's still a `git checkout -b $BRANCH` below, which I think you don't want in either case ($BRANCH being empty or not)

Reply:
I think it's best to add `--no-track` to this `git checkout -B`, even though you're using `git push -u` later.